File: src/main/java/com/vehicletrackingsystem/service/OwnerService.java
package com.vehicletrackingsystem.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.vehicletrackingsystem.domain.Owner;
import com.vehicletrackingsystem.dto.OwnerDTO;
import com.vehicletrackingsystem.dto.OwnerSearchDTO;
import com.vehicletrackingsystem.dto.OwnerPageDTO;
import com.vehicletrackingsystem.dto.OwnerConvertCriteriaDTO;
import com.vehicletrackingsystem.service.GenericService;
import com.vehicletrackingsystem.dto.common.RequestDTO;
import com.vehicletrackingsystem.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface OwnerService extends GenericService<Owner, Integer> {

	List<Owner> findAll();

	ResultDTO addOwner(OwnerDTO ownerDTO, RequestDTO requestDTO);

	ResultDTO updateOwner(OwnerDTO ownerDTO, RequestDTO requestDTO);

    Page<Owner> getAllOwners(Pageable pageable);

    Page<Owner> getAllOwners(Specification<Owner> spec, Pageable pageable);

	ResponseEntity<OwnerPageDTO> getOwners(OwnerSearchDTO ownerSearchDTO);
	
	List<OwnerDTO> convertOwnersToOwnerDTOs(List<Owner> owners, OwnerConvertCriteriaDTO convertCriteria);

	OwnerDTO getOwnerDTOById(Integer ownerId);







}

File: vehicletrackingsystem_ui/src/services/OwnerService.js
import http from "../http-common";

class OwnerService {
  getAllOwners(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/owner/owners`, searchDTO);
  }

  get(ownerId) {
    return this.getRequest(`/owner/${ownerId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/owner?field=${matchData}`, null);
  }

  addOwner(data) {
    return http.post("/owner/addOwner", data);
  }

  update(data) {
  	return http.post("/owner/updateOwner", data);
  }
  
  uploadImage(data,ownerId) {
  	return http.postForm("/owner/uploadImage/"+ownerId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new OwnerService();
